hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -58,56 +158,280 @@ import Ouroboros.Network.Protocol.LocalTxSubmission.Server (
   localTxSubmissionServerPeer,
  )
 import qualified Ouroboros.Network.Protocol.LocalTxSubmission.Type as LocalTxSubmission
+import Ouroboros.Network.Server.RateLimiting (
+  AcceptedConnectionsLimit (..),
+ )
+import Ouroboros.Network.Socket (
+  SomeResponderApplication (..),
+  withServerNode,
+ )
 import qualified Shelley.Spec.Ledger.API as Ledger
+import Test.Cardano.Ledger.Alonzo.Serialisation.Generators ()
+
+data DirectChainLog
 
 withDirectChain ::
-  IO (Channel IO LByteString) ->
+  -- | Tracer for logging
   Tracer IO DirectChainLog ->
+  -- | Network configuration, defined by the genesis configuration.
+  --
+  -- See also 'defaultEpochSlots' and 'defaultNodeToClientVersionData' for playing
+  -- around.
+  (NodeToClientVersionData, EpochSlots) ->
```

Comment:
Why do we parameterize over this if these are plugged as two constants anyways? I'd suggest to hard-code them prominently (at the start of the module) and as long as we do not make them configurable, keep them as CAF